functions/lib/sendgrid/delete_event.py

```diff
@@ -0,0 +1,141 @@
+import os
+import uuid
+from dataclasses import dataclass
+from firebase_functions import https_fn, options
+from google.protobuf.timestamp_pb2 import Timestamp
+from lib.sendgrid.commons import cents_to_dollars
+from lib.constants import db  
+from lib.logging import Logger  
+from lib.sendgrid.constants import DELETE_EVENT_ORGANISER_EMAIL_TEMPLATE_ID, DELETE_EVENT_ATTENDEE_EMAIL_TEMPLATE_ID, SENDGRID_API_KEY
+from sendgrid import SendGridAPIClient
+from sendgrid.helpers.mail import Mail
+
+@dataclass
+class SendGridDeleteEventRequest:
+    eventId: str
+
+    def __post_init__(self):
+        if not isinstance(self.eventId, str):
+            raise ValueError("Event Id must be provided as a string.")
+
+@https_fn.on_call(
+    cors=options.CorsOptions(
+        cors_origins=["https://www.sportshub.net.au", "*"],
+        cors_methods=["post"]
+    ),
+    region="australia-southeast1"
+)
+def send_email_on_delete_event(req: https_fn.CallableRequest):
+    uid = str(uuid.uuid4())
+    logger = Logger(f"sendgrid_delete_event_logger_{uid}")
+    logger.add_tag("uuid", uid)
+
+    body_data = req.data
+    logger.info("Received delete event request.")
+
+    try:
+        request_data = SendGridDeleteEventRequest(**body_data)
+        logger.info(f"Parsed request data: {request_data}.")
+    except ValueError as v:
+        logger.warning(f"Request body did not contain necessary fields. Error: {v}. Returned status=400")
+        return {"status": 400, "message": "Invalid request data"}
+
+    # Get the deleted event data from the database
+    maybe_event_metadata = db.collection("EventsMetadata").document(request_data.eventId).get()
+    if not maybe_event_metadata.exists:
+        logger.error(f"Unable to find deleted event in EventsMetadata. eventId={request_data.eventId}")
+        return {"status": 400, "message": "Event metadata not found"}
+
+    maybe_delete_event_data = db.collection("DeletedEvents").document(request_data.eventId).get()
+    if not maybe_delete_event_data.exists:
+        logger.error(f"Unable to find deleted event in DeletedEvents. eventId={request_data.eventId}")
+        return {"status": 400, "message": "Deleted event data not found"}
+
+    logger.info(f"Retrieved event data for eventId={request_data.eventId}")
+
+    # Retrieve the event data
+    event_metadata_data = maybe_event_metadata.to_dict()
+    event_delete_data = maybe_delete_event_data.to_dict()
+
+    event_name = event_delete_data.get("name")
+    event_price = event_delete_data.get("price")
+    event_status = event_delete_data.get("isActive")
+    organiser_email = event_delete_data.get("userEmail")
+    organiser_name = event_delete_data.get("userName")  
+    event_date = event_delete_data.get("startDate") 
+    date_string = event_date.strftime("%Y-%m-%d %H")
+    purchaser_map = event_metadata_data.get("purchaserMap", {})
+
+    if event_name is None or event_price is None or organiser_email is None or event_status is None:
```

Comment:
should we be checking for the other variables from the `event_delete_data` variable to ensure they are not null as well 🤔